0007-configuration-management-extension/README.md

```diff
@@ -0,0 +1,120 @@
+---
+name: Configuration Management Extension
+shortname: 7/CMX
+status: raw
+editor: Alexey Shmalko <ashmalko@cybervisiontech.com>
+---
+
+## Introduction
+
+The Configuration Management Extension is a [Kaa Protocol](/0001-kaa-protocol/README.md) extension.
+
+It is intended to manage endpoint configuration distribution.
+
+## Requirements and constraints
+### Problems and possible solutions
+
+1. _The server should know if a configuration has been applied by the endpoint._
+
+   Possible solutions:
+   - The client sends a publish message in a response to the new configuration provided by the server.
+
+     This means the server is initiating a request. That does not work well for HTTP, CoAP, and other protocols that only support client-initiated requests.
+
+   - Introduce `/applied` resource, so the client sends an additional request once configuration is applied.
+
+2. _An endpoint might be interested in a part of the general configuration only._
+
+    Possible solutions:
+    - Allow subscribing to a part of configuration. (e.g., using [JSONPath](http://goessner.net/articles/JsonPath/), or JSON Pointer defined in [RFC 6901](https://tools.ietf.org/html/rfc6901).)
+
+3. _Difference between subsequent configurations might be small, so sending the whole configuration is inefficient._
+
+    Possible solutions:
+    - Send only updates. (e.g., using JSON Patch format, defined in [RFC 6902](https://tools.ietf.org/html/rfc6902).)
+
+## Use cases
+
+### UC1: Configuration request
+Configuration delivery by request. The endpoint should be able to receive latest configuration from CMX by request.
+
+### UC2: Configuration push
+Configuration delivery is initiated by the server. The endpoint should receive latest configuration when it connects to the server if this configuration wasn't applied yet.
+
+## Design
+The extension follows client-initiated request/response pattern described in 1/KP, which nicely maps to both MQTT and CoAP.
```

Comment:
I suggest documenting server-initiated requests for MQTT and leaving CoAP design as an open item. It's not designed well anyways. This will avoid pointless breaking of derivative RFCs.